src/comgr.cpp

```diff
@@ -60,6 +60,13 @@
 /// With base driver 5.11.32 the errors disappear.
 /// More info at https://github.com/ROCm/MIOpen/issues/1257.
 #define WORKAROUND_ISSUE_1257 (HIP_PACKAGE_VERSION_FLAT >= 4003021331ULL)
+/// For now, use only standard <limits> to avoid possibility of
+/// correctnes or performance regressions.
+/// \todo Test and enable "custom" local implementation.
+/// Hence the workaround is applicable for a range of hipRTC versions
+/// and potentially in the future
+#define WORKAROUND_HIPRTC_LIMITS \
+    (HIP_PACKAGE_VERSION_FLAT >= 6000023464ULL && HIP_PACKAGE_VERSION_FLAT < 6000023494ULL)
```

Comment:
For reviewers (@atamazov), these way points are the best we know at the moment, they may not be exactly precise but should work with official releases.

Reply:
The HIP version in Staging is usually incorrect. Therefore, we should not rely on these numbers in the code (otherwise we will have more and more of mess in the source code). It's much better to replace the incorrect HIP version numbers with the _correct expected_ ones (in this case 6.1.24xxx) and use the correct version.

Reply:
@atamazov here is a summary of compiler versions and reported in the cmake output:

- [Docker A] ROCm 6.0: 
  - roc-6.0.0 23483 7208e8d15fbf218deb74483ea8c549c67ca4985e
  -  -- Build with HIP 6.0.23494
- [Docker B] The docker with all proposed hipRTC changes:
  -  roc-6.0.0 23483 7208e8d15fbf218deb74483ea8c549c67ca4985e
  - -- Build with HIP 6.0.23494
- [Docker C] The latest mainline staging docker:
  -  amd-mainline-open 23512 8f0cb360b515076885176a3e5ba09dab6ecefa20
  - -- Build with HIP 6.0.23464

There are a few critical issues:
- From Docker A and Docker B: `hipRTC` versioning is now decoupled from `HIP` package, so our code path control needs to accomodate.
- From Docker C: the reported HIP version is incorrect